x-pack/plugins/infra/public/components/asset_details/asset_details.stories.tsx

```diff
@@ -5,23 +5,51 @@
  * 2.0.
  */
 
-import React from 'react';
-import { EuiButton, EuiCard } from '@elastic/eui';
-import { I18nProvider } from '@kbn/i18n-react';
+import React, { useEffect, useState } from 'react';
+import { EuiButton, EuiFlexGroup, EuiSwitch, type EuiSwitchEvent, EuiFlexItem } from '@elastic/eui';
 import type { Meta, Story } from '@storybook/react/types-6-0';
 import { i18n } from '@kbn/i18n';
-import { DecorateWithKibanaContext } from './asset_details.story_decorators';
+import { useArgs } from '@storybook/addons';
+import { DecoratorFn } from '@storybook/react';
 import { AssetDetails } from './asset_details';
 import { decorateWithGlobalStorybookThemeProviders } from '../../test_utils/use_global_storybook_theme';
 import { FlyoutTabIds, type AssetDetailsProps } from './types';
+import { DecorateWithKibanaContext } from './__stories__/decorator';
 
-export default {
-  title: 'infra/Asset Details View/Asset Details Embeddable',
+const links: AssetDetailsProps['links'] = ['apmServices', 'uptime'];
+
+const AssetDetailsDecorator: DecoratorFn = (story) => {
+  const [_, updateArgs] = useArgs();
+  const [checked, setChecked] = React.useState(true);
+
+  useEffect(() => {
+    if (checked) {
+      updateArgs({ links });
+    } else {
+      updateArgs({ links: [] });
+    }
+  }, [updateArgs, checked]);
+
+  const onChange = (e: EuiSwitchEvent) => {
+    setChecked(e.target.checked);
+  };
```

Comment:
nit: this looks like an event consequence and not a side effect, you could simplify it with:
```ts
const onChange = (e: EuiSwitchEvent) => {
  setChecked(e.target.checked);
  updateArgs({ links: e.target.checked ? links : [] });
};
```

Reply:
Yeah. I used `useEffect` here to reset the state when switching between `Page` and `Flyout`. Otherwise, the switch component wouldn't always reflect the `links` args when first rendering these stories